packages/web3-core-method/src/index.js

```diff
@@ -883,7 +895,16 @@ function _handleTxPricing(method, tx) {
                         maxFeePerGas = tx.gasPrice;
                         delete tx.gasPrice;
                     } else {
-                        maxPriorityFeePerGas = tx.maxPriorityFeePerGas || '0x9502F900'; // 2.5 Gwei
+                        maxPriorityFeePerGas =  tx.maxPriorityFeePerGas || '0x9502F900'; // 2.5 Gwei
+
+                        // if no priority fee is set in the tx try to get it from the rpc
+                        if(maxPriorityFeePerGas === '0x9502F900') {
+                            const feeHistory = await getFeeHistory(1);
```

Comment:
There can be optimized [maxPriorityFeePerGas ](https://brg8.medium.com/how-to-build-a-gas-fee-estimator-using-eip-1559-64b3fc9e42ec) calculation, and users can always calculate and set it explicitly via `tx.maxPriorityFeePerGas `. We always defaults to 2.5 Gwei if user is not setting it.

Reply:
I would second @jdevcs comment, we should not make these changes. We can develop a feature to inject custom fee calculation algorithm to the library but in some 4.x releases.

Reply:
personally, I think the hardcoded value of 2.5 gwei is a mistake because the chains differ and as we see on polygon mainnet the transactions fail because of underpricing.
The same thing would from my point of view also apply to the gas calculation which could be done by the user and set with `maxFeePerGas` but this is done in this library to make the usage easier.
If you don't agree then please close the pull request and we make the change for all of our clients

Reply:
As there can be several ways of estimation of `maxPriorityFeePerGas` based on how fast user want his tx to include in next block. So instead of defaulting to single way of estimation in this function, another path could be adding a standalone function for `maxPriorityFeePerGas` estimation for users facilitation, and allowing users to use that function for explicitly doing estimation and setting tx.maxPriorityFeePerGas @mathewmeconry ?

Reply:
@mathewmeconry  our team is occupied for [4.x](https://github.com/ChainSafe/web3.js/tree/4.x) work so if you have time could you go ahead and add maxPriorityFeePerGas estimation function? Thanks

Reply:
@jdevcs sure. in which package should this function live?
so I am gonna remove this improvement again and only add a function to return the estimated priority fee?

Reply:
Yes, `web3.eth` package, once its added in 1.x could you also add this in 4.x for future use.

Reply:
so I guess we can close this pull request cause the `eth_feeHistory` is already exported by the `web3-eth`  module
https://github.com/ChainSafe/web3.js/blob/2c0324af2da467ee1acb72452f20000e916e4306/packages/web3-eth/src/index.js#L426
?